api/client.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
```

Comment:
0/5, maybe split it into client_agent.go and client_coordinator.go

Reply:
We can do that.